H-SSRS

rthedin opened the pull request:
Merging Dave's code with the heuristic model

DaveBrandes committed:
reorganized example config

DaveBrandes committed:
reorganized example config

DaveBrandes committed:
reorganized example config

DaveBrandes committed:
reorganized example config

DaveBrandes committed:
changed the way presence is plotted for HSSRS

DaveBrandes committed:
added more args to H_eagle_track, made changes to accom local moves

DaveBrandes committed:
reduced border on random thermal field

DaveBrandes committed:
minor modifications

DaveBrandes committed:
minor modifications

DaveBrandes committed:
completed local movement action - still to be tested

DaveBrandes committed:
config for local movements around totw

DaveBrandes committed:
example config for totw local moves

DaveBrandes committed:
slight changes to plotting tracks with HSSRS

DaveBrandes committed:
minor modifications

DaveBrandes committed:
changed magn of allow dir change, added some random variation in windfield

DaveBrandes committed:
added simple heuristic model

DaveBrandes committed:
various changes for simple heurisitic model and local moves model

rthedin committed:
Merge branch 'dev' into feature/HSSRS

Conflicts:
	ssrs/layers.py
	ssrs/movmodel.py
	ssrs/simulator.py

rthedin committed:
Clean conflict markers left behind

rthedin committed:
Add plotting parameter back to config

Removed when manually merging with conflict

ewquon committed:
Add kwarg for thermaling decay coefficient

ewquon committed:
Add rand_step_range param to local_moves_mixedlift

ewquon committed:
Add wind perturb params to thermal_soar_glide_local

ewquon committed:
Add new params as kwargs in thermal_soar_glide_local

* `step_wt=0.5`
* `steps_per_circle=8` -- to calculate change in angle per soaring step
* `circling_steps_range=(32, 96)`, random value within this range -- fixed inconsistency with previous hard-coded max steps being 160 @ line #202
* `soaring_radius_range=(20., 60.)`, random value within this range
* `gliding_range=(500., 2000)`, min/max

ewquon committed:
Add back in legacy rulesets for reference

ewquon reviewed (approved):
Nice to see how this has progressed. Did some code cleanup to put tunable parameters into the function keyword arguments list.

rthedin committed:
Fix issues with the HSSRS. See extended message.

* Fix problems related to the two resolutions `resolution` and
 `resolution_terrain`
* Fix PAM format handling with implemented movement model.

rthedin committed:
Fix plotting of single updraft field with `enumerate`

rthedin committed:
Merge branch 'feature/HSSRS' of github.com:rthedin/SSRS into feature/HSSRS

rthedin committed:
Add updated example file for simple heuristics case

rthedin committed:
Progress commit on debugging HSSRS

DaveBrandes committed:
updated config changes

DaveBrandes committed:
fixed conflict

DaveBrandes committed:
added lines to remove nan from wo array with EVVE method

DaveBrandes committed:
removed comment lines added during debugging, renamed a few things

DaveBrandes committed:
made a few changes to plotting

DaveBrandes committed:
removed comment lines added during debugging

DaveBrandes committed:
case example for EVVE paper

rthedin committed:
Integrates H-SSRS and thermal updraft models. See extended commit message

* Add option for wind data source (either `wtk` or `hrrr`)
* Add option for thermal updraft model (either `naive` or `improvedAllen`). Improved Allen uses HRRR data
* Fix handling of multi-variable regex within HRRR class
* Fixed several details on the `compute_thermals_3d` that were broken since major refactoring of the HRRR class
* Re-write of `get_convective_velocity`
* Started a `checkInputs` function to handle incompatible input options (e.g. `improvedAllen` and `wtk`)
* Added function to plot several atmospheric conditions from HRRR (right now, recomputing them)

Still needs to be done:
* Clean up commented out print statements
* Add more checks for new options to ensure the selected options are
* compatible with one another
* Check the height used in HRRR's `wind_velocity_direction_at_altitude` for heights above 100 m
* Save quantities interpolated onto grid from HRRR instead of re-doing it

rthedin committed:
Update example for heuristics over PA ridge with different models

rthedin committed:
Add height information to case ID string when using improved orog model

rthedin committed:
Clean up commented out print statements throughout

rthedin committed:
Fix winddir used on Sx calculation depending on the data source used

rthedin committed:
Add extra checks; round angle for Sx computation

rthedin committed:
Add handling of unspecified thermal realization for no thermal model

rthedin committed:
Fix `None` thermal model specification

rthedin committed:
Adjust error handling for low boundary layer heights

rthedin committed:
Fix issue with tracks not leaving the domain with heuristics model

Tracks would accumulate at the boundary and presence maps would heavily
affected by that. The issue is that `break` only leaves the inner-most
for loop.

rthedin committed:
Small leftovers changes for HSSRS and EVVE

rthedin committed:
Merge branch 'dev' into feature/HSSRS
